libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/guidance/restarea/api/MapboxRestAreaApi.kt

```diff
@@ -0,0 +1,190 @@
+package com.mapbox.navigation.ui.maps.guidance.restarea.api
+
+import com.mapbox.api.directions.v5.models.BannerComponents
+import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.common.ResourceLoadError
+import com.mapbox.common.ResourceLoadResult
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.base.trip.model.roadobject.reststop.RestStop
+import com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer
+import com.mapbox.navigation.ui.maps.guidance.restarea.RestAreaAction
+import com.mapbox.navigation.ui.maps.guidance.restarea.RestAreaProcessor
+import com.mapbox.navigation.ui.maps.guidance.restarea.RestAreaResult
+import com.mapbox.navigation.ui.maps.guidance.restarea.model.GuideMapError
+import com.mapbox.navigation.ui.maps.guidance.restarea.model.GuideMapValue
+import com.mapbox.navigation.ui.maps.guidance.restarea.model.MapboxRestAreaOptions
+import com.mapbox.navigation.ui.utils.internal.resource.ResourceLoaderFactory
+import com.mapbox.navigation.ui.utils.internal.resource.load
+import com.mapbox.navigation.utils.internal.InternalJobControlFactory
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.launch
+
+/**
+ * Mapbox Rest Area Api allows you to generate service area and parking area information
+ * for select maneuvers.
+ * @property accessToken String
+ * @property options MapboxRestAreaOptions
+ */
+class MapboxRestAreaApi @JvmOverloads constructor(
+    val accessToken: String,
+    val options: MapboxRestAreaOptions = MapboxRestAreaOptions.Builder().build()
+) {
+
+    private companion object {
+        private const val ACCESS_TOKEN_QUERY = "?access_token="
```

Comment:
Can we be sure that map URLs will never have any query parameters? If they did, appending `?` would break the URL.

Let's make this more robust and use `?` if there are no query params and `&` if there already are some.

Reply:
> Can we be sure that map URLs will never have any query parameters

You mean URLs will never end with `?` or `&`?

> Let's make this more robust and use ? if there are no query params and & if there already are some.
This would then be applicable to `Signboards` and `Junctions` as well. I am not fully convinced that this check is required, since we Nav API is sure that URLs will look like they do today and are not gonna change. 

@LukasPaczos

Reply:
I think we should employ a little bit of defensive programming and assume that we don't know how the base URL looks like when we're appending a new query parameter.